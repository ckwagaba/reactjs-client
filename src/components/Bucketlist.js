import React from 'react';
import { Link, withRouter } from 'react-router-dom';
import Button from './Button';
import { BASEURL } from '../Config.js';

const Bucketlist = (props) => {
  // redirect to bucketlist form
  const callBucketlistForm = (bucketlistName, bucketlistId) => {
    props.history.push('/bucketlistform/' + bucketlistId + '/' + bucketlistName);
  }

  // handle bucketlist delete
  const handleDelete = (bucketlistId) => {
    fetch(BASEURL + '/bucketlists/' + props.bucketlistId, {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json',
        'Authorization': localStorage.getItem('ACCESSTOKEN')
      }
    })
    .then(response => {
      return response.json()
    })
    .then(responseData => {
      // on successful delete: re-render BucketlistView component
      props.getBucketlists();
    });
  }

  return (
    <div className="bucketlist">
      <Link to={"/bucketlists/" + props.bucketlistId + "/items/"} className="bucketlist_thumbnail">
        {props.bucketlistName}
      </Link>
      <div className="bucketlist_actions">
        <Button buttonText="Edit" className="edit_button" onClick={() => callBucketlistForm(props.bucketlistName, props.bucketlistId)} />
        <Button buttonText="Delete" className="delete_button" onClick={() => handleDelete(props.bucketlistId)} />
      </div>
    </div>
  );
}

export default withRouter(Bucketlist);
