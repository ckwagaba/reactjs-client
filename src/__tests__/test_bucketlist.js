import React from 'react';
import { shallow } from 'enzyme';
import Bucketlist from '../components/bucketlist';

describe('Bucketlist component', () => {

  const component = <Bucketlist />;

  it('should exist', () => {
    expect(component).toMatchSnapshot();
  });

});
