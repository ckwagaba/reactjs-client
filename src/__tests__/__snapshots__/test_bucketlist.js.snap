// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bucketlist component should exist 1`] = `<withRouter(Bucketlist) />`;
